netsetup/desktop_bus.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2019, CESAR. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import sys
+import logging
+import struct
+import dbus
+from dbus.service import method as dbus_method
+
+logging.basicConfig(format='[%(levelname)s] %(funcName)s: %(message)s\n',
+                    stream=sys.stderr, level=logging.INFO)
+
+OPENTHREAD_IFACE = 'br.org.cesar.knot.netsetup.Openthread'
+
+wpantun = None
+
+
+class FailedException(dbus.exceptions.DBusException):
+    _dbus_error_name = 'br.org.cesar.knot.netsetup.Openthread.Error.Failed'
+
+
+class NotSupportedException(dbus.exceptions.DBusException):
+    _dbus_error_name = 'br.org.cesar.knot.netsetup.Openthread.Error.NotSupported'
+
+
+class OpenthreadBus(dbus.service.Object):
+    """
+    br.org.cesar.knot.netsetup.Openthread interface implementation
+    """
+
+    def __init__(self, bus, wpan):
+        self.path = '/br/org/cesar/knot/netsetup/openthread'
+        dbus.service.Object.__init__(self, bus, self.path)
+
+    def __fmt_to_bytes(fmt, arg):
+        """
+            Convert the values in fmt to bytes.
+            The formats strings can be found in 
+            https://docs.python.org/2/library/struct.html#format-strings
+        """
+        return struct.pack(fmt, arg)
+
+    def __bytes_to_dbus_byte_array(arg):
+        """
+            Convert python Bytes in dbus bytes array
+        """
+        return dbus.Array(dbus.Byte(i) for i in arg)
+
+    @dbus_method(dbus.PROPERTIES_IFACE,
+                 in_signature='ss', out_signature='v')
+    def Get(self, interface_name, property_name):
+        """
+            Get property method
+        """
+        return self.GetAll(interface_name)[property_name]
+
+    @dbus_method(dbus.PROPERTIES_IFACE,
+                 in_signature='s', out_signature='a{sv}')
+    def GetAll(self, interface_name):
+        """
+            Get all properties method
+        """
+        global wpantun
```

Comment:
remove the use of `global`

Reply:
Is better to remove global or remove wpan parameter in OpenthreadBus and use global?

Reply:
I prefer the parameter because it keeps the dependency from OpenthreadBus and wpantund.
Besides this...
Why did you separate dbusDesktop and OpenthreadBus?

Reply:
Because in the future can exist another dbus interface, like WifiDbus...

Reply:
Ack